omnibus/files/private-chef-ctl-commands/open_source_chef12_upgrade.rb

```diff
@@ -359,21 +359,33 @@ def make_dir(dir, permissions)
     Dir.mkdir(dir, permissions) unless File.directory?(dir)
   end
 
-  # TODO(jmink) Add error handling
+  # Any config not specified on the command line
+  # should be pulled from /etc/chef-server/chef-server-running.json
   def pull_chef11_db_credentials
-    # This code pulled from knife-ec-backup and adapted
-    log "Pulling open source Chef 11 database credentials"
-    if !File.exists?("/etc/chef-server/chef-server-running.json")
-      log "Failed to find /etc/chef-server/chef-server-running.json"
-      exit 1
-    end
+    sql_host = @options.sql_host
+    sql_port = @options.sql_port
+    sql_pass = @options.sql_pass
+    sql_user = @options.sql_user
 
-    running_config = JSON.parse(File.read("/etc/chef-server/chef-server-running.json"))
-    sql_host = running_config['chef_server']['postgresql']['vip']
-    sql_port = running_config['chef_server']['postgresql']['port']
-    sql_user = running_config['chef_server']['postgresql']['sql_user']
-    sql_password = running_config['chef_server']['postgresql']['sql_password']
-    [sql_host, sql_port, sql_user, sql_password]
+    if !(sql_host && sql_port && sql_pass && sql_user)
+      log "Reading SQL configuration from /etc/chef-server/chef-server-running.json"
+      begin
+        running_config = JSON.parse(File.read("/etc/chef-server/chef-server-running.json"))
+        sql_host ||= running_config['chef_server']['postgresql']['vip']
```

Comment:
Right, the current behavior is such that you can have a mix of command line options or config file options with the command line taking precedence.  I think at the time my thinking was that someone might want to use the running json file to handle the credentials which are a bit large to pass on the command line.  However, I can easily change it to be all-or-nothing.